nucypher/network/nodes.py

```diff
@@ -1036,22 +1037,20 @@ def validate_worker(self, verify_staking: bool = True) -> None:
                           f"from worker {self.worker_address} for stamp {bytes(self.stamp).hex()} "
                 raise self.InvalidWorkerSignature(message)
 
-            # On-chain staking check
-            if verify_staking:
-                if not self._worker_is_bonded_to_staker():  # <-- Blockchain CALL
+            # On-chain staking check, if registry is present
+            if registry:
+                if not self._worker_is_bonded_to_staker(registry=registry):  # <-- Blockchain CALL
                     message = f"Worker {self.worker_address} is not bonded to staker {self.checksum_address}"
                     raise self.DetachedWorker(message)
 
-                if self._staker_is_really_staking():  # <-- Blockchain CALL
+                if self._staker_is_really_staking(registry=registry):  # <-- Blockchain CALL
```

Comment:
This comment "<-- Blockchain CALL" kinda says to me that we could probably figure out a solid naming scheme we can use for methods/functions that make blockchain calls so it's inherent in the name and not the comment.